daffodil-core/src/main/scala/org/apache/daffodil/dsom/ElementBase.scala

```diff
@@ -342,7 +343,12 @@ trait ElementBase
           // will work.
           //
           val str = defaultValueAsString
-          val value = primType.fromXMLString(str)
+          val value = try {
+            primType.fromXMLString(str)
+          } catch {
+            case ipd: InvalidPrimitiveDataException =>
+              SDE("Invalid default value: %s".format(str, primType, ipd.getMessage))
```

Comment:
I think SDE takes a string and varargs. So you don't call format yourself, it does that lazily when it computes the actual message.

Reply:
There's also only a single %s in the format string, but 3 format parameters. I think you just want ipd.getMessage.

Reply:
Yes, fixed.